programs/bpf_loader/src/lib.rs

```diff
@@ -107,11 +107,16 @@ macro_rules! deploy_program {
      $account_size:expr, $slot:expr, $drop:expr, $new_programdata:expr $(,)?) => {{
         let mut load_program_metrics = LoadProgramMetrics::default();
         let mut register_syscalls_time = Measure::start("register_syscalls_time");
-        let deployment_program_runtime_environment = create_program_runtime_environment_v1(
-            &$invoke_context.feature_set,
-            $invoke_context.get_compute_budget(),
+        let deployment_slot: Slot = $slot;
+        let environments = $invoke_context.get_runtime_environments_for_slot(
+            deployment_slot.saturating_add(DELAY_VISIBILITY_SLOT_OFFSET)
+        ).map_err(|e| {
+            ic_msg!($invoke_context, "Failed to get runtime environment: {}", e);
```

Comment:
I would add a comment here (and in the corresponding -v4 code) that says that the error case here can never happen, as the schedule is guaranteed to be set by this point